### ISSUE TITLE
Source address option

### ISSUE THREAD

**Issue by lars18th** [rofl0r/rrredir#1]
Hi,

I'm sure this tool uses the KISS principle. However, I suggest to add a new "simple" option:

- Include support for selecting the SOURCE address. Example:
```
rrredir [-b -i listenip -p port -t timeout -s address] ip1:port1 ip2:port2...
```

- or best at destination level. Example:
```
rrredir [-b -i listenip -p port -t timeout] ip1:port1@address ip2:port2@address...
```

The objective is to select wich one of the multiple IP addresses (loopback, wan, lan, etc.) is used to connect to the destination.

Please, consider it.
Thank you!

**Comment by rofl0r** [comment]
hi, i've implemented variant 1 via `-b bindaddr`.
if it works as expected, i will implement the second variant too.
please test (branch proper_bind)!

**Comment by lars18th** [comment]
> hi, i've implemented variant 1 via `-b bindaddr`.
> if it works as expected, i will implement the second variant too.
> please test (branch proper_bind)!

I'll test it soon.
Thank you!

**Comment by lars18th** [comment]
Hi @rofl0r ,

> please test (branch proper_bind)!

Done! It works! :+1: 
Thank you.

> if it works as expected, i will implement the second variant too.

I suggest to use this syntax as example:
`rrredir -i 127.0.0.1 -p 3128 -b 10.0.0.30 192.168.3.27:8080 -b 192.168.1.50 172.16.1.31:8080`

That's signifies "multiple `-b` parameters", and use the _last_ indicated when a new target appears in the command-line.
You agree?

**Comment by rofl0r** [comment]
> That's signifies "multiple -b parameters", and use the last indicated when a new target appears in the command-line.

i prefered your previous suggestion target:port@bindaddr
it seems less confusing, and (i didnt test yet) multiple parameters with same value could be problematic with getopt

**Comment by lars18th** [comment]
OK, @rofl0r 

However, in this case you need to describe what signifies to use both, the `-b` parameter and the `@bindaddr`. I feel you like to implement in this way: `-b` is used by default if no `@bindaddr` is present, and `@bindaddr` overwrites the default. Right?

**Comment by rofl0r** [comment]
yeah. -b will mean "use this bindaddr for *all* outgoing connections" and using it together with @... will result in error.

**Comment by lars18th** [comment]
> yeah. -b will mean "use this bindaddr for _all_ outgoing connections" and using it together with @... will result in error.

This is one solution.

The other is to use "-b" as default (replacing the system default) and replace it _only_ when some @ is used.
For example, this will be valid:
`rrredir -i 127.0.0.1 -p 8080 -t 5 -b 192.168.1.30 10.0.20.1:8080@192.168.3.30 192.168.100.10:8080 
 192.168.200.10:8080 172.16.0.30:8080@192.168.3.30`
And it uses the upstreams "192.168.100.10" & "192.168.100.10" with the "binded 192.168.1.30".

You agree?

**Comment by rofl0r** [comment]
yeah, that seems to be a reasonable behaviour.

**Comment by rofl0r** [comment]
implemented in PR #4. would you be so kind as to test it, @lars18th ?

**Comment by lars18th** [comment]
> implemented in PR #4. would you be so kind as to test it, @lars18th ?

:+1: Working perfectly. Thank you!
I feel it's time to merge it with the mainstream.

Regards.

**Comment by lars18th** [comment]
Hi @rofl0r ,

Thank you for this upgrade. Good job!

However, remember to update the README with the command line parameters:
```
rrredir [-i listenip -p port -t timeout -b bindaddr] ip1:port1@binaddr ip2:port2@binaddr...
```

Regards.
